docs/src/main/asciidoc/security-openid-connect-client.adoc

```diff
@@ -548,13 +552,28 @@ public interface ProtectedResourceService {
 
 Alternatively, `AccessTokenRequestFilter` can be registered automatically with all MP Rest or JAX-RS clients if `quarkus.oidc-token-propagation.register-filter` property is set to `true` and `quarkus.oidc-token-propagation.json-web-token` property is set to `false` (which is a default value).
 
-This filter will be additionally enhanced in the future to support exchanging the access tokens before propagating them.
+==== Exchange Token Before Propagation
 
-=== JsonWebTokenRequestFilter
+If the current access token needs to be exchanged before propagation and you work with link:https://github.com/keycloak/keycloak-documentation/blob/master/securing_apps/topics/token-exchange/token-exchange.adoc[Keycloak] or other OpenId Connect Provider which supports a link:https://tools.ietf.org/html/rfc8693[Token Exchange] token grant then you can configure `AccessTokenRequestFilter` like this:
+
+[source,properties]
+----
+quarkus.oidc-client.auth-server-url=http://localhost:8180/auth/realms/quarkus
+quarkus.oidc-client.client-id=quarkus-app
+quarkus.oidc-client.credentials.secret=secret
+quarkus.oidc-client.grant.type=exchange
+quarkus.oidc-client.grant-options.exchange.requested_subject=bob
```

Comment:
I would avoid introducing right now delegation but consider only impersonation use cases. Where the subject of the original token is still the authorized party and the service is only interested in exchanging to a new token meant to access some downstream service.

IMO, this is the first use case we should consider.

Reply:
@pedroigor Hi Pedro, sure, I'll just update the docs to highlight the impersonation scenario - it would work for the delegation as well since it is a map - the client just collects all properties from this map with the matching grant name :-)